### PR TITLE
TF-2983 Fix download EML with special character

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- #2983 

## Demo
Screenshot 2024-09-25 at 10 37 19

### CODE REVIEW

File: model/lib/email/attachment.dart [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Attachment with EquatableMixin {
       'name' : name ?? '',
       'type' : type?.mimeType ?? '',
     });
-    return Uri.decodeFull(downloadUri);
+    return downloadUri;
```
Comment (dab246) [inline model/lib/email/attachment.dart:52]: Please write unit test for it.

**Review by dab246 — COMMENTED** [review]
- [ ] Test again  download attachment

File: model/lib/email/attachment.dart (hunk 1 above)
Comment (tddang-linagora) [reply model/lib/email/attachment.dart:52]: Done

**Review by dab246 — APPROVED** [review]

**Review by dab246 — COMMENTED** [review]
- [x] Please rebase it

File: model/test/email/attachment_test.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:model/email/attachment.dart';
+
+void main() {
+  final accountId = AccountId(Id('1'));
+  const baseDownloadUrl = 'http://localhost/download/{accountId}/{blobId}/{name}?name={name}&type={type}';
+
+  group('attachment test:', () {
+    group('getDownloadUrl:', () {
```
Comment (hoangdat) [inline model/test/email/attachment_test.dart:12]: please write more about the case 
- name include ? (Fx: `Subject has question mark ?`)
- Unicode name ? (Fx: `Đây là tiêu đề thư`)
- ...

Comment (tddang-linagora) [reply model/test/email/attachment_test.dart:12]: Done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**